apps/api/src/domain/models/tag/tag.entity.ts

```diff
@@ -55,6 +57,27 @@ export class Tag extends Aggregate implements HasLabel {
         });
     }
 
+    addMember(taggedMemberCompositeIdentifier: CategorizableCompositeIdentifier) {
+        /**
+         * Note that this is catching what would eventually be an invariant
+         * validation error sooner rather than later.
```

Comment:
Put some thought into whether this is duplication of logic or whether "failing fast" is more important.